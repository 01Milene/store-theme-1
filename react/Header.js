import React, { Component } from 'react'
import PropTypes from 'prop-types'
import { injectIntl, intlShape } from 'react-intl'
import Input from '@vtex/styleguide/lib/Input'
import Button from '@vtex/styleguide/lib/Button'

class Header extends Component {
  static propTypes = {
    name: PropTypes.string,
    intl: intlShape.isRequired,
  }

  translate = id => this.props.intl.formatMessage({ id: `dreamstore.${id}` })

  render() {
    const { name } = this.props
    return (
      <div className="flex justify-between items-center w-100 top-0 pa4 pa5-l bg-light-gray">
        {name && <div className="b f3 near-black mt3">{name}</div>}
        <div className="mb0">
          <Input long placeholder={this.translate('search-placeholder')} />{' '}
          <Button>{this.translate('search')}</Button>
        </div>
      </div>
    )
  }
}

export default injectIntl(Header)
