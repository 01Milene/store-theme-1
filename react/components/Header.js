import React, { Component } from 'react'
import PropTypes from 'prop-types'
import { injectIntl, intlShape } from 'react-intl'
import Input from '@vtex/styleguide/lib/Input'
import Button from '@vtex/styleguide/lib/Button'

import CartIcon from '../images/CartIcon'

class Header extends Component {
  constructor(props) {
    super(props)
    this.state = {
      searchValue: '',
    }
  }

  static propTypes = {
    name: PropTypes.string,
    intl: intlShape.isRequired,
  }

  translate = id => this.props.intl.formatMessage({ id: `dreamstore.${id}` })

  handleChange = ({ target: { value } }) => {
    this.setState({ searchValue: value })
  }

  handleSearch = () => location.assign(`/${this.state.searchValue}/s`)

  handleCart = () => location.assign('/checkout/#/cart')

  render() {
    const { account } = global.__RUNTIME__
    const { name } = this.props
    const { searchValue } = this.state
    return (
      <div className="z-2 flex-ns justify-between items-center w-100 top-0 pa4 pa5-ns bg-white bb bw1 b--serious-black tc tl-ns">
        <a className="link b f3 near-black tc tl-ns" href="/">
          {name || account}
        </a>
        <div className="tr-ns flex items-center">
          <Input
            long
            placeholder={this.translate('search-placeholder')}
            value={searchValue}
            onChange={this.handleChange}
          />{' '}
          <div className="mt3 flex items-center justify-center">
            <Button onClick={this.handleSearch}>
              {this.translate('search')}
            </Button>
            <Button onClick={this.handleCart}>
              <CartIcon />
            </Button>
          </div>
        </div>
      </div>
    )
  }
}

export default injectIntl(Header)
