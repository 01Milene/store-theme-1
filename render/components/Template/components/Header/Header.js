import Link from 'vtex.render/Link'
import React, {Component, PropTypes} from 'react'

import Meta from './components/Meta'
import User from './components/User'
import Menu from 'vtex.nav-menu'
import Minicart from './components/Minicart'
import SearchBar from './components/SearchBar'

// eslint-disable-next-line
class Header extends Component {
  constructor (props) {
    super(props)
    this.state = {isNavMenuOpen: false}
  }

  handleClick = () => {
    this.setState({isNavMenuOpen: !this.state.isNavMenuOpen})
  }

  render () {
    return (
      <header className="relative w-100 shadow-1 bg-white">
        <Meta customTags={this.props.customTags} title={this.props.title} />
        <div className="mw9-ns center flex flex-wrap justify-start items-center ph2-ns pv2-ns">
          <div className="content-start dn-ns">
            <button className="bn pointer dim pv3 pr3 pl2 relative bg-transparent" onClick={this.handleClick}>
              <svg className="v-mid" height="24" viewBox="0 0 24 24" width="24" xmlns="http://www.w3.org/2000/svg">
                <path className="fill-light-primary" d="M3 18h18v-2H3v2zm0-5h18v-2H3v2zm0-7v2h18V6H3z" />
              </svg>
            </button>
          </div>
          <div className="pl1-ns">
            <Link className="db w4 w-70-m w-100-l link dim light-primary" to="/">
              <span className="dn f4 f3-ns brand-name">Dreamstore</span>
              <img src="//dreamstore.vteximg.com.br/arquivos/logo-vtex.png" />
            </Link>
          </div>
          <div className="flex-auto w-100 w-auto-l order-last order-0-l ph2 pb2 ph0-ns pb1-m pv2-l">
            <SearchBar
              searchTerm={this.props.searchTerm}
              placeholder="Buscar"
            />
          </div>
          <User />
          <Minicart />
        </div>
        <Menu
          items={[
            {label: 'Moda', href: '/moda/c'},
            {label: 'EletrodomÃ©sticos', href: '/eletrodomesticos/c'},
            {label: 'Perfumaria', href: '/perfumaria/c'},
          ]}
          isMobileMenuOpen={this.state.isNavMenuOpen}
          onToggleMenu={this.handleClick}
        />
      </header>
    )
  }
}

Header.propTypes = {
  customTags: PropTypes.arrayOf(PropTypes.object),
  searchTerm: PropTypes.string,
  title: PropTypes.string,
}

export default Header
