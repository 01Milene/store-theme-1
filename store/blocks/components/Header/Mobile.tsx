import { HeaderLayout, HeaderRow, HeaderSpacer } from 'vtex.store-header'
import { Logo } from 'vtex.store-components'
import { Menu } from 'vtex.menu'
import { Login } from 'vtex.login'
import { Drawer } from 'vtex.store-drawer'

import MenuItemClothing from './Menu/Items/Clothing'
import MenuItemDecoration from './Menu/Items/Decoration'
import MenuItemCustomSale from './Menu/Items/CustomSale'
import Minicart from './Minicart'

export default function Mobile() {
  return (
    <HeaderLayout.Mobile __id="">
      <HeaderRow __id="1-mobile">
        <Drawer __id="">
          <Menu __id="drawer" orientation="vertical">
            <MenuItemClothing />
            <MenuItemDecoration />
            <MenuItemCustomSale />
          </Menu>
        </Drawer>
        <Logo
          __id="mobile"
          title="Logo"
          href="/"
          url="https://storecomponents.vteximg.com.br/arquivos/store-theme-logo-mobile.png"
          width="70"
        />
        <HeaderSpacer __id={null} />
        <Login __id={null} />
        <Minicart />
      </HeaderRow>
    </HeaderLayout.Mobile>
  )
}
