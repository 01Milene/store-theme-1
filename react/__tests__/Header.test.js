import React from 'react'
import { mountWithIntl, loadTranslation, setLocale } from 'enzyme-react-intl'
import Header from '../components/Header'

describe('Header test', () => {
  let wrapperPT

  beforeEach(() => {
    global.__RUNTIME__ = { account: 'account' }
    loadTranslation('../locales/pt-BR.json')
    setLocale('pt-BR')
    wrapperPT = mountWithIntl(<Header />)
  })

  it('should have 3 div elements', () => {
    expect(wrapperPT.find('div').length).toBe(3)
  })

  it('should render correctly pt-BR', () => {
    expect(wrapperPT).toMatchSnapshot()
  })

  it('should render correctly en-US', () => {
    loadTranslation('../locales/en-US.json')
    setLocale('en-US')
    const wrapperEN = mountWithIntl(<Header />)
    expect(wrapperEN).toMatchSnapshot()
  })

  it('should render correctly es-AR', () => {
    loadTranslation('../locales/es-AR.json')
    setLocale('es-AR')
    const wrapperES = mountWithIntl(<Header />)
    expect(wrapperES).toMatchSnapshot()
  })
})
