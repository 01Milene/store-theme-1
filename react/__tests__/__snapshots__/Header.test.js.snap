// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Header test should render correctly en-US 1`] = `
<InjectIntl(Header)
  intl={
    Object {
      "defaultFormats": Object {},
      "defaultLocale": "en",
      "formatDate": [Function],
      "formatHTMLMessage": [Function],
      "formatMessage": [Function],
      "formatNumber": [Function],
      "formatPlural": [Function],
      "formatRelative": [Function],
      "formatTime": [Function],
      "formats": Object {},
      "formatters": Object {
        "getDateTimeFormat": [Function],
        "getMessageFormat": [Function],
        "getNumberFormat": [Function],
        "getPluralFormat": [Function],
        "getRelativeFormat": [Function],
      },
      "locale": "en-US",
      "messages": Object {
        "dreamstore.buy": "Buy",
        "dreamstore.cart": "Cart",
        "dreamstore.search": "Search",
        "dreamstore.search-placeholder": "Search for products, brands...",
        "dreamstore.soon": "Soon...",
      },
      "now": [Function],
      "textComponent": "span",
    }
  }
>
  <Header
    intl={
      Object {
        "defaultFormats": Object {},
        "defaultLocale": "en",
        "formatDate": [Function],
        "formatHTMLMessage": [Function],
        "formatMessage": [Function],
        "formatNumber": [Function],
        "formatPlural": [Function],
        "formatRelative": [Function],
        "formatTime": [Function],
        "formats": Object {},
        "formatters": Object {
          "getDateTimeFormat": [Function],
          "getMessageFormat": [Function],
          "getNumberFormat": [Function],
          "getPluralFormat": [Function],
          "getRelativeFormat": [Function],
        },
        "locale": "en-US",
        "messages": Object {
          "dreamstore.buy": "Buy",
          "dreamstore.cart": "Cart",
          "dreamstore.search": "Search",
          "dreamstore.search-placeholder": "Search for products, brands...",
          "dreamstore.soon": "Soon...",
        },
        "now": [Function],
        "textComponent": "span",
      }
    }
  >
    <div
      className="flex-ns justify-between items-center w-100 top-0 pa2 pa5-l bg-light-gray"
    >
      <a
        className="link b f3 near-black mt3"
        href="/"
      >
        account
      </a>
      <div
        className="tr-ns flex"
      >
        <Input
          autoFocus={false}
          disabled={false}
          error={false}
          long={true}
          multiple={false}
          onChange={[Function]}
          placeholder="Search for products, brands..."
          readOnly={false}
          token={false}
          value=""
        >
          <label
            className="dib"
            style={
              Object {
                "width": "350px",
              }
            }
          >
            <span
              className="db mb3 "
            />
            <input
              autoFocus={false}
              className="w-100 ma0 border-box bw1 br2 b--solid outline-0 near-black b--light-gray hover-b--silver bg-white f6 pv3 ph5 "
              disabled={false}
              multiple={false}
              onBlur={[Function]}
              onChange={[Function]}
              onFocus={[Function]}
              placeholder="Search for products, brands..."
              readOnly={false}
              value=""
            />
          </label>
        </Input>
         
        <div
          className="mt3"
        >
          <Button
            autoFocus={false}
            block={false}
            disabled={false}
            icon={false}
            large={false}
            onClick={[Function]}
            primary={false}
            secondary={false}
            type="button"
            xLarge={false}
          >
            <button
              autoFocus={false}
              className="bw1 ba fw5 ttu br2 fw4 pv3 ph5 f6 b--transparent blue bg-transparent hover-heavy-blue hover-b--transparent pointer "
              disabled={false}
              onClick={[Function]}
              type="button"
            >
              Search
            </button>
          </Button>
          <Button
            autoFocus={false}
            block={false}
            disabled={false}
            icon={false}
            large={false}
            onClick={[Function]}
            primary={true}
            secondary={false}
            type="button"
            xLarge={false}
          >
            <button
              autoFocus={false}
              className="bw1 ba fw5 ttu br2 fw4 pv3 ph5 f6 b--blue bg-blue white hover-bg-heavy-blue hover-b--heavy-blue pointer "
              disabled={false}
              onClick={[Function]}
              type="button"
            >
              Cart
            </button>
          </Button>
        </div>
      </div>
    </div>
  </Header>
</InjectIntl(Header)>
`;

exports[`Header test should render correctly es-AR 1`] = `
<InjectIntl(Header)
  intl={
    Object {
      "defaultFormats": Object {},
      "defaultLocale": "en",
      "formatDate": [Function],
      "formatHTMLMessage": [Function],
      "formatMessage": [Function],
      "formatNumber": [Function],
      "formatPlural": [Function],
      "formatRelative": [Function],
      "formatTime": [Function],
      "formats": Object {},
      "formatters": Object {
        "getDateTimeFormat": [Function],
        "getMessageFormat": [Function],
        "getNumberFormat": [Function],
        "getPluralFormat": [Function],
        "getRelativeFormat": [Function],
      },
      "locale": "es-AR",
      "messages": Object {
        "dreamstore.buy": "Comprar",
        "dreamstore.cart": "Carrito",
        "dreamstore.search": "Buscar",
        "dreamstore.search-placeholder": "Búsqueda por productos, marcas...",
        "dreamstore.soon": "Breve...",
      },
      "now": [Function],
      "textComponent": "span",
    }
  }
>
  <Header
    intl={
      Object {
        "defaultFormats": Object {},
        "defaultLocale": "en",
        "formatDate": [Function],
        "formatHTMLMessage": [Function],
        "formatMessage": [Function],
        "formatNumber": [Function],
        "formatPlural": [Function],
        "formatRelative": [Function],
        "formatTime": [Function],
        "formats": Object {},
        "formatters": Object {
          "getDateTimeFormat": [Function],
          "getMessageFormat": [Function],
          "getNumberFormat": [Function],
          "getPluralFormat": [Function],
          "getRelativeFormat": [Function],
        },
        "locale": "es-AR",
        "messages": Object {
          "dreamstore.buy": "Comprar",
          "dreamstore.cart": "Carrito",
          "dreamstore.search": "Buscar",
          "dreamstore.search-placeholder": "Búsqueda por productos, marcas...",
          "dreamstore.soon": "Breve...",
        },
        "now": [Function],
        "textComponent": "span",
      }
    }
  >
    <div
      className="flex-ns justify-between items-center w-100 top-0 pa2 pa5-l bg-light-gray"
    >
      <a
        className="link b f3 near-black mt3"
        href="/"
      >
        account
      </a>
      <div
        className="tr-ns flex"
      >
        <Input
          autoFocus={false}
          disabled={false}
          error={false}
          long={true}
          multiple={false}
          onChange={[Function]}
          placeholder="Búsqueda por productos, marcas..."
          readOnly={false}
          token={false}
          value=""
        >
          <label
            className="dib"
            style={
              Object {
                "width": "350px",
              }
            }
          >
            <span
              className="db mb3 "
            />
            <input
              autoFocus={false}
              className="w-100 ma0 border-box bw1 br2 b--solid outline-0 near-black b--light-gray hover-b--silver bg-white f6 pv3 ph5 "
              disabled={false}
              multiple={false}
              onBlur={[Function]}
              onChange={[Function]}
              onFocus={[Function]}
              placeholder="Búsqueda por productos, marcas..."
              readOnly={false}
              value=""
            />
          </label>
        </Input>
         
        <div
          className="mt3"
        >
          <Button
            autoFocus={false}
            block={false}
            disabled={false}
            icon={false}
            large={false}
            onClick={[Function]}
            primary={false}
            secondary={false}
            type="button"
            xLarge={false}
          >
            <button
              autoFocus={false}
              className="bw1 ba fw5 ttu br2 fw4 pv3 ph5 f6 b--transparent blue bg-transparent hover-heavy-blue hover-b--transparent pointer "
              disabled={false}
              onClick={[Function]}
              type="button"
            >
              Buscar
            </button>
          </Button>
          <Button
            autoFocus={false}
            block={false}
            disabled={false}
            icon={false}
            large={false}
            onClick={[Function]}
            primary={true}
            secondary={false}
            type="button"
            xLarge={false}
          >
            <button
              autoFocus={false}
              className="bw1 ba fw5 ttu br2 fw4 pv3 ph5 f6 b--blue bg-blue white hover-bg-heavy-blue hover-b--heavy-blue pointer "
              disabled={false}
              onClick={[Function]}
              type="button"
            >
              Carrito
            </button>
          </Button>
        </div>
      </div>
    </div>
  </Header>
</InjectIntl(Header)>
`;

exports[`Header test should render correctly pt-BR 1`] = `
<InjectIntl(Header)
  intl={
    Object {
      "defaultFormats": Object {},
      "defaultLocale": "en",
      "formatDate": [Function],
      "formatHTMLMessage": [Function],
      "formatMessage": [Function],
      "formatNumber": [Function],
      "formatPlural": [Function],
      "formatRelative": [Function],
      "formatTime": [Function],
      "formats": Object {},
      "formatters": Object {
        "getDateTimeFormat": [Function],
        "getMessageFormat": [Function],
        "getNumberFormat": [Function],
        "getPluralFormat": [Function],
        "getRelativeFormat": [Function],
      },
      "locale": "pt-BR",
      "messages": Object {
        "dreamstore.buy": "Comprar",
        "dreamstore.cart": "Carrinho",
        "dreamstore.search": "Buscar",
        "dreamstore.search-placeholder": "Busque por produtos, marcas...",
        "dreamstore.soon": "Em breve...",
      },
      "now": [Function],
      "textComponent": "span",
    }
  }
>
  <Header
    intl={
      Object {
        "defaultFormats": Object {},
        "defaultLocale": "en",
        "formatDate": [Function],
        "formatHTMLMessage": [Function],
        "formatMessage": [Function],
        "formatNumber": [Function],
        "formatPlural": [Function],
        "formatRelative": [Function],
        "formatTime": [Function],
        "formats": Object {},
        "formatters": Object {
          "getDateTimeFormat": [Function],
          "getMessageFormat": [Function],
          "getNumberFormat": [Function],
          "getPluralFormat": [Function],
          "getRelativeFormat": [Function],
        },
        "locale": "pt-BR",
        "messages": Object {
          "dreamstore.buy": "Comprar",
          "dreamstore.cart": "Carrinho",
          "dreamstore.search": "Buscar",
          "dreamstore.search-placeholder": "Busque por produtos, marcas...",
          "dreamstore.soon": "Em breve...",
        },
        "now": [Function],
        "textComponent": "span",
      }
    }
  >
    <div
      className="flex-ns justify-between items-center w-100 top-0 pa2 pa5-l bg-light-gray"
    >
      <a
        className="link b f3 near-black mt3"
        href="/"
      >
        account
      </a>
      <div
        className="tr-ns flex"
      >
        <Input
          autoFocus={false}
          disabled={false}
          error={false}
          long={true}
          multiple={false}
          onChange={[Function]}
          placeholder="Busque por produtos, marcas..."
          readOnly={false}
          token={false}
          value=""
        >
          <label
            className="dib"
            style={
              Object {
                "width": "350px",
              }
            }
          >
            <span
              className="db mb3 "
            />
            <input
              autoFocus={false}
              className="w-100 ma0 border-box bw1 br2 b--solid outline-0 near-black b--light-gray hover-b--silver bg-white f6 pv3 ph5 "
              disabled={false}
              multiple={false}
              onBlur={[Function]}
              onChange={[Function]}
              onFocus={[Function]}
              placeholder="Busque por produtos, marcas..."
              readOnly={false}
              value=""
            />
          </label>
        </Input>
         
        <div
          className="mt3"
        >
          <Button
            autoFocus={false}
            block={false}
            disabled={false}
            icon={false}
            large={false}
            onClick={[Function]}
            primary={false}
            secondary={false}
            type="button"
            xLarge={false}
          >
            <button
              autoFocus={false}
              className="bw1 ba fw5 ttu br2 fw4 pv3 ph5 f6 b--transparent blue bg-transparent hover-heavy-blue hover-b--transparent pointer "
              disabled={false}
              onClick={[Function]}
              type="button"
            >
              Buscar
            </button>
          </Button>
          <Button
            autoFocus={false}
            block={false}
            disabled={false}
            icon={false}
            large={false}
            onClick={[Function]}
            primary={true}
            secondary={false}
            type="button"
            xLarge={false}
          >
            <button
              autoFocus={false}
              className="bw1 ba fw5 ttu br2 fw4 pv3 ph5 f6 b--blue bg-blue white hover-bg-heavy-blue hover-b--heavy-blue pointer "
              disabled={false}
              onClick={[Function]}
              type="button"
            >
              Carrinho
            </button>
          </Button>
        </div>
      </div>
    </div>
  </Header>
</InjectIntl(Header)>
`;
